op-batcher/batcher/driver.go

```diff
@@ -302,6 +397,26 @@ func (l *BatchSubmitter) loop() {
 				continue
 			}
 			l.publishStateToL1(queue, receiptsCh)
+		case targetDAType := <-economicDATypeCh:
+			l.lastStoredBlock = eth.BlockID{}
+			// close current state to prepare for switch
+			err := l.state.Close()
+			if err != nil {
+				if errors.Is(err, ErrPendingAfterClose) {
+					l.Log.Warn("Closed channel manager to handle DA type switch with pending channel(s) remaining - submitting")
+				} else {
+					l.Log.Error("Error closing the channel manager to handle a DA type switch", "err", err)
+				}
+			}
+			// on DA type switch we want to publish all pending state then wait until each result clears before resetting
+			// the state.
+			publishAndWait()
+			l.clearState(l.shutdownCtx)
+			// switch action after clear state
+			l.switchDAType(targetDAType)
+			time.Sleep(time.Minute) // wait op-node derivation published DA data
```

Comment:
I had doubts about why the code here needs to sleep for 1 minute. Through private communication, I understood the purpose and documented it here.
After switching the type of DA data, op-node cannot immediately consume the old type of DA data, causing op-batcher to resubmit DA data of the new type, resulting in duplicate data submission and waste of funds. Therefore, after op-batcher switches the type of DA data, it needs to wait for 1 minute to give op-node enough time to consume all the old type data.